### PR TITLE
fcrypt, crypto, gpgme: refactor shared gpgagent shutdown code

### PR DESCRIPTION
**Pull request by petermax2**
## Basics

Do not describe the purpose here but:

- [x] Short descriptions should be in the release notes (added as entry in
      doc/news/_preparation_next_release.md which contains `*(my name)*`)
      **Please always add something to the the release notes.**
- [ ] Longer descriptions should be in documentation or in design decisions.
- [x] Describe details of how you changed the code in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, should be in the commit messages.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests
- [x] I ran all tests locally and everything went fine
- [ ] affected documentation is fixed
- [ ] I added code comments, logging, and assertions (see doc/CODING.md)
- [ ] meta data is updated (e.g. README.md of plugins)

## Review

Remove the line below and add the "work in progress" label if you do
not want the PR to be reviewed:

@markus2330 please review my pull request

### CODE REVIEW

**Comment by markus2330** [comment]
Thank you for looking into this!

@ingwinlu any idea what the build server had here?

File: src/plugins/crypto/gpgagent_teardown.c [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/**
+ * @file
+ *
+ * @brief common method for shuttind down the gpg-agent in unit tests
```
Comment (markus2330) [inline src/plugins/crypto/gpgagent_teardown.c]: Typo: shuttind

File: src/plugins/crypto/gpgagent_teardown.c [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ *
+ */
+
+static void test_teardown (void)
```
Comment (markus2330) [inline src/plugins/crypto/gpgagent_teardown.c]: You could also make a .h file and have a "static inline" function.

**Comment by petermax2** [comment]
thank you for the feedback!

**Comment by markus2330** [comment]
Thank you, nicely done!

**Pull request merged**